### ISSUE TITLE
Dio error response always return isRedirect false

### ISSUE THREAD

**Issue by pedromassango** [flutterchina/dio#821]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   Android   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |    8.0  |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |   3.0.9   |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |   IntelliJ 2020.1   |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Dio error response always return `isRedirect` false even if `DioError.response.redirects` list is not empty.

**Comment by pedromassango** [comment]
Don't stale

**Comment by wendux** [comment]
I test with  dio-3.0.9,  it worked!
```
var response = await Dio().get("http://flutterchina.club");
  print(response.isRedirect);  // true
```

**Comment by pedromassango** [comment]
So this can be closed then. I'm not unsing Dio actually
